realm-transformer/src/main/kotlin/io/realm/transformer/RealmTransformer.kt

```diff
@@ -0,0 +1,471 @@
+/*
+ * Copyright 2018 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.transformer
+
+import com.android.SdkConstants
+import com.android.build.api.transform.*
+import com.google.common.io.Files
+import io.realm.annotations.RealmClass
+import javassist.ClassPool
+import javassist.CtClass
+import javassist.CtField
+import javassist.NotFoundException
+import javassist.bytecode.ClassFile
+import org.gradle.api.Project
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.util.jar.JarFile
+import java.util.regex.Pattern
+
+/**
+ * This class implements the Transform API provided by the Android Gradle plugin.
+ */
+class RealmTransformer(val project: Project) : Transform() {
```

Comment:
If you carry on, you might even end up fixing https://github.com/realm/realm-java/issues/3971 😄